src/Raven.Server/Documents/Subscriptions/AbstractSubscriptionStorage.cs

```diff
@@ -393,6 +407,102 @@ public void LowMemory(LowMemorySeverity lowMemorySeverity)
         }
     }
 
+    protected static void SetSubscriptionHistory(AbstractSubscriptionConnectionsState subscriptionConnectionsState, SubscriptionDataAbstractBase subscriptionData)
+    {
+        subscriptionData.RecentConnections = subscriptionConnectionsState.RecentConnections;
+        subscriptionData.RecentRejectedConnections = subscriptionConnectionsState.RecentRejectedConnections;
+        subscriptionData.CurrentPendingConnections = subscriptionConnectionsState.PendingConnections;
+    }
+
+    protected SubscriptionState GetRunningSubscriptionInternal(ClusterOperationContext context, long? id, string name, bool history, out TState connectionsState)
+    {
+        SubscriptionState state;
+        if (string.IsNullOrEmpty(name) == false)
+        {
+            state = GetSubscriptionFromServerStore(context, name);
+        }
+        else if (id.HasValue)
+        {
+            name = GetSubscriptionNameById(context, id.Value);
+            state = GetSubscriptionFromServerStore(context, name);
+        }
+        else
+        {
+            throw new ArgumentNullException("Must receive either subscription id or subscription name in order to provide subscription data");
+        }
+
+        if (_subscriptions.TryGetValue(state.SubscriptionId, out connectionsState) == false)
+            return null;
+
+        if (connectionsState.IsSubscriptionActive() == false)
+            return null;
+
+        return state;
+    }
+
+    public virtual SubscriptionState GetSubscription(ClusterOperationContext context, long? id, string name, bool history)
```

Comment:
I do not like it, it's either GetSubscriptionById or GetSubscriptionByName, should not be both at the same time

Reply:
Also we on purpose started naming methods 'FromServerStore' to distinguish from where the value comes, let's keep the nomenclature

Reply:
Applies to other methods as well

Reply:
done